fix(nuxt): deprecate scanning directory index plugins (#18418)
nuxt/nuxt: docs/2.guide/2.directory-structure/1.plugins.md

```bash
plugins
 | - myPlugin.ts  // scanned

>  | - mySecondPlugin.ts  // scanned

Reviewer: ```suggestion
```

 | - myOtherPlugin
 | --- supportingFile.ts   // not scanned
 | --- componentToRegister.vue   // not scanned

>  | --- index.ts  // not scanned

Reviewer: ```suggestion
 | --- index.ts  // currently scanned but deprecated
```

```

> Only `myPlugin.ts` and `mySecondPlugin.ts` would be registered. You can configure [`plugins`](/docs/api/configuration/nuxt-config#plugins-1) to include unscanned files.

Reviewer: ```suggestion
Only `myPlugin.ts` and `myOtherPlugin/index.ts` would be registered. You can configure [`plugins`](/docs/api/configuration/nuxt-config#plugins-1) to include unscanned files.
```
